### PR TITLE
Fix score calculation error when intersecting FsaVec and Fsa on cpu

### PR DESCRIPTION
**Pull request by pkufool**
See https://github.com/k2-fsa/k2/pull/759#issuecomment-874357866

### CODE REVIEW

File: k2/csrc/fsa_algo.cu [hunk 1]
```diff
@@ -203,13 +203,17 @@ bool Intersect(FsaOrVec &a_fsas, int32_t properties_a, FsaOrVec &b_fsas,
     int32_t this_num_arcs = creator.GetArcOffsetFor(i + 1) - arc_offset;
     if (arc_map_a) {
       int32_t arc_offset_a = a_fsas_row_splits12_data[i * stride_a];
-      for (int32_t i = 0; i < this_num_arcs; i++)
-        this_arc_map_a[i] += arc_offset_a;
+      for (int32_t i = 0; i < this_num_arcs; i++) {
```
Comment (csukuangfj) [inline k2/csrc/fsa_algo.cu:206]: Could you also add a test to cover it?

File: k2/csrc/fsa_algo.cu [hunk 2]
```diff
@@ -203,13 +203,17 @@ bool Intersect(FsaOrVec &a_fsas, int32_t properties_a, FsaOrVec &b_fsas,
...
+        this_arc_map_a[i] = this_arc_map_a[i] == -1 ? this_arc_map_a[i] :
```
Comment (csukuangfj) [inline k2/csrc/fsa_algo.cu:207]: I would suggest
Screen Shot 2021-07-06 at 3 50 38 PM
which is clearer, I think. You can keep it if you like it.

File: k2/csrc/fsa_algo.cu (hunk 1 above)
Comment (pkufool) [reply k2/csrc/fsa_algo.cu:206]: Ok

**Comment by pkufool** [comment]
@csukuangfj  can this PR merge now?

**Comment by csukuangfj** [comment]
Thanks!

**Pull request merged**